src/components/CreateNewPin.js

```diff
@@ -8,25 +13,178 @@ import {
+  const [pinName, setPinName] = useState("");
+  const [pinDescription, setPinDescription] = useState("");
+  const [pinLink, setPinLink] = useState("");
+  const [pinImage, setPinImage] = useState();
+  const [isGetPinImage, setIsGetPinImage] = useState(false);
+  const [selectedFile, setSelectedFile] = useState();
+  const [preview, setPreview] = useState();
+  const [isPinCreated, setIsPinCreated] = useState(false);
+
+  const redirect = useNavigate();
+
+  async function handleImageUpload(e) {
+    const imageFile = e.target.files[0];
+
+    // show image preview
+    if (!e.target.files || e.target.files.length === 0) {
+      setSelectedFile(undefined);
+
+      return;
+    }
+    setSelectedFile(imageFile);
+
+    // compress image
+    console.log(`original size ${imageFile.size / 1024 / 1024} MB`);
+    const imageName = imageFile.name.split(".");
+    localStorage.setItem("uploadedImageName", imageName[0]);
+
+    const options = {
+      maxSizeMB: 1,
+      maxWidthOrHeight: 1920,
+      useWebWorker: true,
+    };
+    try {
+      const compressedFile = await imageCompression(imageFile, options);
+      console.log(`compressed size ${compressedFile.size / 1024 / 1024} MB`);
+
+      // convert pinImage from Blob to string and store in localStorage
+      const reader = new FileReader();
+      reader.onload = (event) => {
+        localStorage.setItem("uploadedImage", event.target.result);
+      };
+      reader.readAsDataURL(compressedFile);
+      setIsGetPinImage(true);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  const dataURLtoBlob = (dataURL) => {
+    // convert base64 to raw binary data held in a string
+    const byteString = atob(dataURL.split(",")[1]);
+    // separate out the mime component
+    let mimeString = dataURL.split(",")[0].split(":")[1].split(";")[0];
+    // write the bytes of the string to an ArrayBuffer
+    let arrayBuffer = new ArrayBuffer(byteString.length);
+    let _ia = new Uint8Array(arrayBuffer);
+    for (let i = 0; i < byteString.length; i++) {
+      _ia[i] = byteString.charCodeAt(i);
+    }
+
+    var dataView = new DataView(arrayBuffer);
+    var blob = new Blob([dataView], {type: mimeString});
+    console.log(`uploading size ${blob.size / 1024 / 1024} MB`);
+
+    return blob;
+  };
 
-  const submitPinData = () => {
+  const submitPinData = (dataURLtoBlob) => {
     if (!pinName || !pinDescription || !pinLink) {
-      alert("please check if all fields are filled~");
-      
+      alert("please check if all fields are filled");
+
+      return;
+    } else if (pinName && pinDescription && pinLink && !pinImage) {
+      alert("please upload and check the image for your pin");
+
       return;
     }
-    // here's the pin-data that will be upload to firebase
-    console.log("pinName", pinName);
-    console.log("pinDescription", pinDescription);
-    console.log("pinLink", pinLink);
+    dataURLtoBlob(localStorage.getItem("uploadedImage"));
```

Comment:
What if 'localStorage.getItem("uploadedImage")' get nothing in local storage? Please handle this.